### PR TITLE
Migrate format API test for playwright

### PR DESCRIPTION
**Pull request by pavanpatil1**
## What?

Part of https://github.com/WordPress/gutenberg/issues/38851.
Migrate [format-api.test.js](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/plugins/format-api.test.js) to its Playwright version.
## Why?

Part of https://github.com/WordPress/gutenberg/issues/38851.
## How?

See[ MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.
## Testing Instructions

Run` npm run test-e2e:playwright test/e2e/specs/editor/plugins/format-api.spec.js`

### CODE REVIEW

**Comment by pavanpatil1** [comment]
Hello @kevin940726, Could you review this PR. ?.

**Comment by pavanpatil1** [comment]
Hi @kevin940726, Hope you are doing well. 
Could you please help me in reviewing this PR ?.

File: test/e2e/specs/editor/plugins/format-api.spec.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		pageUtils,
+	} ) => {
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( 'First paragraph' );
+		await pageUtils.pressKeyWithModifier( 'shiftAlt', 'ArrowLeft' );
+		await editor.clickBlockToolbarButton( 'More' );
+		const button = await page.locator(
+			`//button[contains(text(), '${ 'Custom Link' }')]`
+		);
+		await button.click();
```
Comment (kevin940726) [inline test/e2e/specs/editor/plugins/format-api.spec.js]: Have you read the [best practice](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/README.md#use-accessible-selectors) guide? I think we can refactor this into role selector instead:

```suggestion
		await page.click( 'role=button[name="Custom Link"i]' );
```

**Review by kevin940726 — COMMENTED** [review]
Hi! Sorry I missed the pings! Thank you as always 💯

**Comment by pavanpatil1** [comment]
Hi @kevin940726, Yes, I tried that, but because the button doesn't have the name attribute available, it is failing. 

image



screenshot-nimbusweb me-2022 07 08-07_43_41

**Comment by kevin940726** [comment]
@pavanpatil1 It's not a DOM attribute but a computed accessible attribute interpreted by the browsers and screen readers. You can view what accessible attributes browsers see via devtools, take Chrome as an example:

image

You can see that the element has a role of `menuitem` and a name of ` Custom Link`. The `` part is likely caused by a bug and should probably be fixed by a different PR. For now, I think we can just use the selector `role=menuitem[name=/Custom Link/i]` to target the element (and add a comment above explaining why we want to use regex here).

**Comment by pavanpatil1** [comment]
Hi @kevin940726, Thank you so much 💯 

I addressed the above feedback. Could you check again ?.

**Review by kevin940726 — COMMENTED** [review]
Looks good! Remember to also delete the snapshot of the original test though (`format-api.test.js.snap`).

**Review by kevin940726 — APPROVED** [review]
LGTM 👍 Thanks! 💯

**Pull request merged**